File: python/dmp_bbo/task_solver.py
## \file task_solver.py
#  \author Freek Stulp
#  \brief  TaskSolver class
#  
#  \ingroup Demos
#  \ingroup DMP_BBO

## Interface for classes that can perform rollouts.
# For further information see the section on \ref sec_bbo_task_and_task_solver
class TaskSolver:
    
    ## Perform rollouts, that is, given a set of samples, determine all the variables that are relevant to evaluating the cost function. 
    # \param[in] sample The sample
    # \return The variables relevant to computing the cost.
    def performRollout(self,sample):
        raise NotImplementedError('subclasses must override performRollout()!')
        
    def plotRollout(self,cost_vars,ax):
        print("plotRollout not implemented.")

File: src/dmp_bbo/TaskSolver.cpp
/**
 * @file   TaskSolver.cpp
 * @brief  TaskSolver class source file.
 * @author Freek Stulp
 *
 * This file is part of DmpBbo, a set of libraries and programs for the 
 * black-box optimization of dynamical movement primitives.
 * Copyright (C) 2014 Freek Stulp, ENSTA-ParisTech
 * 
 * DmpBbo is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Lesser General Public License as published by
 * the Free Software Foundation, either version 2 of the License, or
 * (at your option) any later version.
 * 
 * DmpBbo is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Lesser General Public License for more details.
 * 
 * You should have received a copy of the GNU Lesser General Public License
 * along with DmpBbo.  If not, see <http://www.gnu.org/licenses/>.
 */

#include "dmp_bbo/TaskSolver.hpp"

#include "dmpbbo_io/EigenFileIO.hpp"

#include <iostream>
#include <iomanip>
#include <fstream>
#include <boost/filesystem.hpp>
#include <eigen3/Eigen/Core>

using namespace std;
using namespace Eigen;

namespace DmpBbo {
  
TaskSolver::TaskSolver(void)
{
}

TaskSolver::~TaskSolver(void)
{
}

void TaskSolver::performRollout(const Eigen::VectorXd& sample, Eigen::MatrixXd& cost_vars) const
{
  Eigen::VectorXd task_parameters;
  performRollout(sample,task_parameters,cost_vars);
};


}
